Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -100,17 +104,53 @@ public bool IsAllowed(EntityUid user, EntityUid target, AccessReaderComponent? r
         if (!reader.Enabled)
             return true;
 
+        var validIds = GetAllowedEntities(user, reader).ToList();
```

Comment:
Why is this before the checks? I could be wrong, but wont this act weird if the user is in a box? Your validIds are for the box, not the user inside the box.

Reply:
Did it by mistake and weirdly it works when it should not. Cardboard boxes do this weird double accessing when bumping into doors. And there is an event GetAdditionalAccessEvent. Can't figure it out at all so I'm giving up on this PR.

Reply:
You did a good job so far I wouldn't give up! Just ask in the discord if you need more help.e